content/en/blog/_posts/2019-12-09-volume-snapshot-beta.md

```diff
@@ -0,0 +1,449 @@
+API Version:  snapshot.storage.k8s.io/v1beta1
+Kind:         VolumeSnapshot
+Metadata:
+  Creation Timestamp:  2019-11-16T00:36:04Z
+  Finalizers:
+    snapshot.storage.kubernetes.io/volumesnapshot-as-source-protection
+    snapshot.storage.kubernetes.io/volumesnapshot-bound-protection
+  Generation:        1
+  Resource Version:  1294
+  Self Link:         /apis/snapshot.storage.k8s.io/v1beta1/namespaces/default/volumesnapshots/new-snapshot-demo
+  UID:               32ceaa2a-3802-4edd-a808-58c4f1bd7869
+Spec:
+  Source:
+    Persistent Volume Claim Name:  test-pvc
+  Volume Snapshot Class Name:      test-snapclass
+Status:
+  Bound Volume Snapshot Content Name:  snapcontent-32ceaa2a-3802-4edd-a808-58c4f1bd7869
+  Creation Time:                       2019-11-16T00:36:04Z
+  Ready To Use:                        true
+  Restore Size:                        1Gi
+```
+
+As a reminder to any developers building controllers using volume snapshot APIs: before using a `VolumeSnapshot` API object, validate the bi-directional binding between the `VolumeSnpashot` and the `VolumeSnapshotContent` it is bound to, to ensure the binding is complete and correct (not doing so may result in security issues).
+
+```shell
+kubectl describe volumesnapshotcontent
+
+Name:         snapcontent-32ceaa2a-3802-4edd-a808-58c4f1bd7869
+Namespace:
+Labels:       <none>
+Annotations:  <none>
+API Version:  snapshot.storage.k8s.io/v1beta1
+Kind:         VolumeSnapshotContent
+Metadata:
+  Creation Timestamp:  2019-11-16T00:36:04Z
+  Finalizers:
+    snapshot.storage.kubernetes.io/volumesnapshotcontent-bound-protection
+  Generation:        1
+  Resource Version:  1292
+  Self Link:         /apis/snapshot.storage.k8s.io/v1beta1/volumesnapshotcontents/snapcontent-32ceaa2a-3802-4edd-a808-58c4f1bd7869
+  UID:               7dfdf22e-0b0c-4b71-9ddf-2f1612ca2aed
+Spec:
+  Deletion Policy:  Delete
+  Driver:           testdriver.csi.k8s.io
+  Source:
+    Volume Handle:             d1b34a5f-0808-11ea-808a-0242ac110003
+  Volume Snapshot Class Name:  test-snapclass
+  Volume Snapshot Ref:
+    API Version:       snapshot.storage.k8s.io/v1beta1
+    Kind:              VolumeSnapshot
+    Name:              test-snapshot
+    Namespace:         default
+    Resource Version:  1286
+    UID:               32ceaa2a-3802-4edd-a808-58c4f1bd7869
+Status:
+  Creation Time:    1573864564608810101
+  Ready To Use:     true
+  Restore Size:     1073741824
+  Snapshot Handle:  127c5798-0809-11ea-808a-0242ac110003
+Events:             <none>
+```
+
+#### Importing an existing volume snapshot with Kubernetes
+
+You can always expose a pre-existing volume snapshot in Kubernetes by manually creating a `VolumeSnapshotContent` object to represent the existing volume snapshot. Because `VolumeSnapshotContent` is a non-namespace API object, only a cluster admin may have the permission to create it. By specifying the `volumeSnapshotRef` the cluster admin specifies exactly which user can use the snapshot.
```

Comment:
nit:
```suggestion
You can always expose a pre-existing volume snapshot in Kubernetes by manually creating a VolumeSnapshotContent object to represent the existing volume snapshot. Because VolumeSnapshotContent is a non-namespace API object, only a cluster admin may have the permission to create it. By specifying the `volumeSnapshotRef` the cluster admin specifies exactly which user can use the snapshot.
```